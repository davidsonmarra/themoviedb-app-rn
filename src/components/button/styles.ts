import {StyleSheet} from 'react-native';
import {colors} from '../../configs/theme';

interface Props {
  isDisabled: boolean;
}

const getStyles = ({isDisabled}: Props) =>
  StyleSheet.create({
    container: {
      justifyContent: 'space-between',
      alignItems: 'center',
      padding: 8,
      borderRadius: 12,
      width: '100%',
      opacity: 1,
    },
    primary: {
      backgroundColor: isDisabled ? colors.disabled : colors.primary,
    },
    secondary: {},
  });

export default getStyles;
