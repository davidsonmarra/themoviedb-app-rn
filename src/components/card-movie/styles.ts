import {StyleSheet} from 'react-native';
import {colors} from '../../configs/theme';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    borderWidth: 1,
    borderColor: colors.borderLight,
    borderRadius: 8,
    marginBottom: 16,
  },
  image: {
    width: 188 / 2,
    height: 282 / 2,
    borderTopLeftRadius: 8,
    borderBottomLeftRadius: 8,
  },
  contentDescription: {
    flex: 1,
    padding: 12,
  },
  cardTitle: {
    fontSize: 16,
    fontWeight: 'bold',
    color: colors.title,
  },
  cardDate: {
    fontSize: 14,
    color: colors.description,
  },
  cardOverview: {
    marginTop: 8,
    fontSize: 14,
    color: colors.text,
  },
});

export default styles;
