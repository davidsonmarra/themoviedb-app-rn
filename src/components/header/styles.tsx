import {StyleSheet} from 'react-native';
import {colors} from '../../configs/theme';

const styles = StyleSheet.create({
  container: {
    width: '100%',
    paddingHorizontal: 12,
    paddingVertical: 16,
    flexDirection: 'row',
    backgroundColor: colors.primary,
  },
  icon: {
    width: 24,
  },
  text: {
    flex: 1,
    textAlign: 'center',
  },
});

export default styles;
