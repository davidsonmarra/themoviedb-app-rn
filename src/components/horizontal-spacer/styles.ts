import {StyleSheet} from 'react-native';

const getStyles = (width: number) => {
  return StyleSheet.create({
    container: {
      width,
    },
  });
};

export default getStyles;
