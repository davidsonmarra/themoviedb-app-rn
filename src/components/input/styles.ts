import {StyleSheet} from 'react-native';
import {colors} from '../../configs/theme';

const styles = StyleSheet.create({
  container: {
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderWidth: 1,
    borderRadius: 5,
    borderColor: colors.border,
    paddingHorizontal: 10,
  },
  input: {
    flex: 1,
    height: 40,
  },
});

export default styles;
