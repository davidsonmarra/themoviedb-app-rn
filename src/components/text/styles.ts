import {StyleSheet} from 'react-native';
import {colors} from '../../configs/theme';

const styles = StyleSheet.create({
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    color: colors.title,
  },
  text: {
    fontSize: 16,
    fontWeight: 'normal',
    color: colors.text,
  },
  textBold: {
    fontSize: 16,
    fontWeight: 'bold',
    color: colors.text,
  },
  buttonPrimary: {
    fontSize: 18,
    fontWeight: 'bold',
    color: colors.buttonText,
    padding: 8,
  },
  buttonSecondary: {
    fontSize: 18,
    fontWeight: 'bold',
    color: colors.primary,
    padding: 8,
  },
  label: {
    fontSize: 14,
    fontWeight: 'normal',
    color: colors.text,
  },
  error: {
    fontSize: 14,
    fontWeight: 'normal',
    color: colors.error,
  },
  header: {
    fontSize: 18,
    fontWeight: 'bold',
    color: colors.title,
  },
  overlayTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    color: colors.buttonText,
  },
  overlayText: {
    fontSize: 16,
    color: colors.buttonText,
  },
});

export default styles;
