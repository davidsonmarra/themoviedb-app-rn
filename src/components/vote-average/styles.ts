import {StyleSheet} from 'react-native';

const getStyles = (color: string) =>
  StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'center',
      width: 64,
      height: 64,
      borderRadius: 32,
      borderWidth: 5,
      borderColor: color,
    },
    average: {
      color,
      fontSize: 28,
    },
  });

export default getStyles;
