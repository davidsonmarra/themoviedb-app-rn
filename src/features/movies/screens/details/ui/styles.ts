import {StyleSheet} from 'react-native';
import {colors} from '../../../../../configs/theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  imageBackground: {
    flex: 1,
  },
  overlay: {
    flex: 1,
    backgroundColor: colors.overlayDark,
    paddingHorizontal: 16,
    paddingVertical: 24,
  },
  title: {
    color: colors.buttonText,
    marginBottom: 8,
    textAlign: 'center',
  },
  containerInfo: {
    flexDirection: 'row',
    marginBottom: 24,
  },
  containerGenres: {
    flexDirection: 'row',
  },
  subtitle: {
    marginBottom: 8,
  },
  average: {
    marginTop: 24,
    marginHorizontal: 'auto',
  },
});

export default styles;
