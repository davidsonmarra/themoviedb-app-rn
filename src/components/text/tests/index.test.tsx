import React from 'react';
import {render, screen} from '@testing-library/react-native';
import Text, {TextType} from '..';
import styles from '../styles';

const mockText = 'Snapshot test!';
const renderComponent = (type: TextType = TextType.title) =>
  render(<Text type={type}>{mockText}</Text>);

describe('Text', () => {
  it('renders correctly', () => {
    const tree = renderComponent().toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders children correctly', () => {
    renderComponent();
    const text = screen.getByRole('text');
    expect(text.children).toEqual([mockText]);
  });

  it('renders title style correctly', () => {
    renderComponent(TextType.title);
    const text = screen.getByRole('text');
    expect(text.props.style[0]).toBe(styles.title);
  });

  it('renders text style correctly', () => {
    renderComponent(TextType.text);
    const text = screen.getByRole('text');
    expect(text.props.style[0]).toBe(styles.text);
  });

  it('renders textBold style correctly', () => {
    renderComponent(TextType.textBold);
    const text = screen.getByRole('text');
    expect(text.props.style[0]).toBe(styles.textBold);
  });

  it('renders buttonPrimary style correctly', () => {
    renderComponent(TextType.buttonPrimary);
    const text = screen.getByRole('text');
    expect(text.props.style[0]).toBe(styles.buttonPrimary);
  });

  it('renders buttonSecondary style correctly', () => {
    renderComponent(TextType.buttonSecondary);
    const text = screen.getByRole('text');
    expect(text.props.style[0]).toBe(styles.buttonSecondary);
  });

  it('renders label style correctly', () => {
    renderComponent(TextType.label);
    const text = screen.getByRole('text');
    expect(text.props.style[0]).toBe(styles.label);
  });

  it('renders error style correctly', () => {
    renderComponent(TextType.error);
    const text = screen.getByRole('text');
    expect(text.props.style[0]).toBe(styles.error);
  });

  it('renders header style correctly', () => {
    renderComponent(TextType.header);
    const text = screen.getByRole('text');
    expect(text.props.style[0]).toBe(styles.header);
  });
});
