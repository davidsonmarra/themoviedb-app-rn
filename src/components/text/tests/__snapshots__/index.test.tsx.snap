// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#121212",
        "fontSize": 24,
        "fontWeight": "bold",
      },
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;
